### DISCUSSION TITLE
Show initial preloader till the server sent html is rendered

### DISCUSSION THREAD

**Discussion by norbertSG** [nuxt/nuxt#25711] [Questions]
I use SSR, I would like to display to the user a preloader/loader till the server sent the initial html to the client, once the client is ready then i would like to remove the preloader.

Somehow insert a html and remove it after the original html is ready.

Something like we have on nuxt development, till the server is built up is displayed that green preloader.

Is that somehow possible, via some hooks or ... ?

@danielroe

**Comment by danielroe** [comment]
You can use `spaLoadingTemplate` (just create a file in `app/spa-loading-template.html`)...

**Reply by norbertSG** [reply]
It works only with ssr: false

https://nuxt.com/blog/v3-6

My app is ssr: true and the initial page render, first time takes around 5sec, because the first api call is slower after that is cached.

I would like to display a loader or a different html till the original html is served from the server.

Is that possible somehow ?

**Reply by danielroe** [reply]
No. Any loader would need to be _in_ the initial HTML served.

You could load your data on the client-side though.
